holidays/financial/national_stock_exchange_of_india.py

```diff
@@ -0,0 +1,187 @@
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from datetime import date
+from gettext import gettext as tr
+from typing import Optional
+
+from holidays.calendars import _CustomHinduHolidays, _CustomIslamicHolidays
+from holidays.calendars.gregorian import (
+    JUN,
+    MAR,
+    JUL,
+    OCT,
+    NOV,
+)
+from holidays.groups.christian import ChristianHolidays
+from holidays.groups.hindu import HinduCalendarHolidays
+from holidays.groups.islamic import IslamicHolidays
+from holidays.observed_holiday_base import SAT_TO_NONE, SUN_TO_NONE, ObservedHolidayBase
+
+BAKRI_ID = "BAKRI_ID"
+CHRISTMAS_DAY = "CHRISTMAS_DAY"
+DIWALI_BALIPRATIPADA = "DIWALI_BALIPRATIPADA"
+DIWALI_LAXMI_PUJAN = "DIWALI_LAXMI_PUJAN"
+DUSSEHRA = "DUSSEHRA"
+DR_BABA_SAHEB_AMBEDKAR_JAYANTI = "DR_BABA_SAHEB_AMBEDKAR_JAYANTI"
+GANESH_CHATURTHI = "GANESH_CHATURTHI"
+GOOD_FRIDAY = "GOOD_FRIDAY"
+GURU_NANAK_JAYANTI = "GURU_NANAK_JAYANTI"
+HOLI = "HOLI"
+ID_UL_FITR = "ID_UL_FITR"
+INDEPENDENCE_DAY = "INDEPENDENCE_DAY"
+MAHARASHTRA_DAY = "MAHARASHTRA_DAY"
+MAHA_SHIVARATRI = "MAHA_SHIVARATRI"
+MAHATMA_GANDHI_JAYANTI = "MAHATMA_GANDHI_JAYANTI"
+MAHAVIR_JAYANTI = "MAHAVIR_JAYANTI"
+MUHARRAM = "MUHARRAM"
+RAM_NAVAMI = "RAM_NAVAMI"
+REPUBLIC_DAY = "REPUBLIC_DAY"
+
+
+class NationalStockExchangeOfIndia(
+    ObservedHolidayBase, HinduCalendarHolidays, ChristianHolidays, IslamicHolidays
+):
+    """National Stock Exchange of India (NSE) holidays.
+
+    References:
+        * <https://web.archive.org/web/20250821175252/https://www.nseindia.com/resources/exchange-communication-circulars>
+
+    Historical data:
+        * [2022](https://web.archive.org/web/20250821071611/https://nsearchives.nseindia.com/content/circulars/CMTR50560.pdf)
+        * [2023](https://web.archive.org/web/20250821071635/https://nsearchives.nseindia.com/content/circulars/CMTR54757.pdf)
+        * [2024](https://web.archive.org/web/20250821071650/https://nsearchives.nseindia.com/content/circulars/CMTR59722.pdf)
+        * [2025](https://web.archive.org/web/20250624132016/https://nsearchives.nseindia.com/content/circulars/CMTR65587.pdf)
+    """
+
+    market = "XNSE"
+    default_language = "en_IN"
+    supported_languages = ("en_IN", "en_US", "hi")
+    estimated_label = tr("%s")
+    # NSE launched its services in 1994.
+    start_year = 1995
+
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault("observed_rule", SAT_TO_NONE + SUN_TO_NONE)
+        HinduCalendarHolidays.__init__(self, cls=NationalStockExchangeOfIndiaHinduHolidays)
+        ChristianHolidays.__init__(self)
+        IslamicHolidays.__init__(self, cls=NationalStockExchangeOfIndiaIslamicHolidays)
+        super().__init__(*args, **kwargs)
+
+    DIWALI_LAXMI_PUJAN_DATES = {
+        2024: (NOV, 1),
+        2025: (OCT, 21),
+    }
+
+    MUHARRAM_DATES = {
+        2024: (JUL, 17),
+    }
```

Comment:
Sort + allowing estimate label for Islamic holidays in year outside 2022-2025
```suggestion
    def __init__(self, *args, islamic_show_estimated: bool = True, **kwargs):
        """
        Args:
            islamic_show_estimated:
                Whether to add "estimated" label to Islamic holidays name
                if holiday date is estimated.
        """
        ChristianHolidays.__init__(self)
        HinduCalendarHolidays.__init__(self, cls=NationalStockExchangeOfIndiaHinduHolidays)
        IslamicHolidays.__init__(
            self,
            cls=NationalStockExchangeOfIndiaIslamicHolidays,
            show_estimated=islamic_show_estimated,
        )
        kwargs.setdefault("observed_rule", SAT_TO_NONE + SUN_TO_NONE)
        super().__init__(*args, **kwargs)
```